packages/flame_bloc/README.md

```diff
@@ -1,53 +1,109 @@
-a `BlocProvider` like this:
+Lets assume we have a bloc that handles player inventory, first we need to make it available to our
+components.
+
+We can do that by using `FlameBlocProvider` component:
 
 ```dart
-BlocProvider<ExampleGame>(
-  create: (_) => InventoryBloc(),
-  child: GameWidget(game: ExampleGame()),
-)
+class MyGame extends FlameGame {
+  @override
+  Future<void> onLoad() async {
+    await add(
+      FlameBlocProvider<PlayerInventoryBloc, PlayerInventoryState>(
+        create: () => PlayerInventoryBloc(),
+        children: [
+          Player(),
+          // ...
+        ],
+      ),
+    );
+  }
+}
 ```
 
-To enable the features of `flame_bloc` in your game, you can make your game class inherit from
-`FlameBlocGame`, or if you are already using an enhanced `FlameGame` class (like for example a
-`Forge2DGame`), the `FlameBloc` mixin can be used instead.
+With that, the `Player` component will now have access to our bloc.
 
-
-To access the bloc from inside your game, the `read` method can be used.
+If more than one bloc needs to be provided, `FlameMultiBlocProvider` can be used in a similar fashion:
 
 ```dart
-class ExampleGame extends FlameBlocGame {
-  void selectWeapon() {
-    read<InventoryBloc>.add(WeaponSelected('axe'));
+class MyGame extends FlameGame {
+  @override
+  Future<void> onLoad() async {
+    await add(
+      FlameMultiBlocProvider(
+        providers: [
+          FlameBlocProvider<PlayerInventoryBloc, PlayerInventoryState>(
+            create: () => PlayerInventoryBloc(),
+          ),
+          FlameBlocProvider<PlayerStatsBloc, PlayerStatsState>(
+            create: () => PlayerStatsBloc(),
+          ),
+        ],
+        children: [
+          Player(),
+          // ...
+        ],
+      ),
+    );
   }
 }
 ```
 
-To have your components listen to state change, the `BlocComponent` mixin can be used.
+Listening to states changes on a component level can be done with two approaches:
 
+By using `FlameBlocListener` component:
 
 ```dart
-class PlayerComponent with BlocComponent<InventoryBloc, InventoryState> {
-
-  // onNewState can be overriden to so the component
-  // can be notified on state changes
+class Player extends PositionComponent {
   @override
-  void onNewState(InventoryState state) {
-    print(state.weapon);
+  Future<void> onLoad() async {
+    await add(
+      FlameBlocListener<PlayerInventoryBloc, PlayerInventoryState>(
+        listener: (state) {
+          updateGear(state);
+        },
+      ),
+    );
   }
+}
+```
+
+Or by using `FlameBlocListenable` mixin:
+
+```dart
+class Player extends PositionComponent
+  with FlameBlocListenable<PlayerInventoryBloc, PlayerInventoryState> {
 
   @override
-  void update(double dt) {
-    // the `state` getter can also be used to have
-    // direct access to the current state
-    print(state.weapon);
+  void onNewState(state) {
+    updateGear(state);
+  }
+}
+```
```

Comment:
Might want to call out one limitation of the mixin is that a component cannot listen to multiple blocs